src/rules/sortObjectLiteralKeysRule.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "unsorted key '";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new SortedKeyWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class SortedKeyWalker extends Lint.RuleWalker {
+    private lastSortedKeyStack: string[] = [];
+    private sortedStateStack: boolean[] = [];
```

Comment:
Sure, can do that, this will of cause the entire recursive object literal check to fail so something like 
```
{
  b: "b",
  a: "a",
  c: {
    "b": "b",
    "a": "a"
  }
}
```

won't throw an error on the inner object literal error. Is that okay?